flink-ai-flow/ai_flow_plugins/scheduler_plugins/airflow/event_handler.py

```diff
@@ -0,0 +1,209 @@
+
+
+class RestartEventHandler(EventHandler):
+    """
+    Internal event handler that always return restart scheduling action.
+    """
+
+    def handle_event(self, event: BaseEvent, task_state: object) -> Tuple[SchedulingAction, object]:
+        return SchedulingAction.RESTART, task_state
+
+
+class ActionEventHandler(EventHandler):
+    """
+    Internal event handler that handle event base on the key of the event.
+    """
+
+    def handle_event(self, event: BaseEvent, task_state: object) -> Tuple[SchedulingAction, object]:
+        if 'stop' == event.key:
+            return SchedulingAction.STOP, task_state
+        elif 'restart' == event.key:
+            return SchedulingAction.RESTART, task_state
+        else:
+            return SchedulingAction.START, task_state
+
+
+class AiFlowTs(object):
+    def __init__(self):
+        # key: namespace event_type sender event_key
+        # value: BaseEvent
+        self.event_map = {}
+        self.schedule_time = 0
+        self.latest_time = 0
+
+    def __str__(self):
+        return "EVENT_MAP: {0} TIME: {1} LATEST: {2}".format(self.event_map, self.schedule_time, self.latest_time)
+
+
+class ActionWrapper(object):
+    def __init__(self, action=SchedulingAction.NONE):
+        self.action = action
+
+
+class AIFlowHandler(EventHandler):
+    """
+    AIFlowHandler is an implementation of EventHandler,
+    which implements the semantics of ai flow based on event scheduling.
+    """
+    def __init__(self, config: str):
+        self.config = config
+
+    @staticmethod
+    def parse_configs(config_str: str):
+        configs: List[ConditionConfig] = []
+        config_json = json.loads(config_str)
+        for config in config_json:
+            condition_config = ConditionConfig(event_key=config['event_key'],
+                                               event_value=config['event_value'],
+                                               event_type=config['event_type'],
+                                               action=TaskAction(config['action']),
+                                               condition_type=ConditionType(config['condition_type']),
+                                               value_condition=ValueCondition(config['value_condition']),
+                                               life=EventLife(config['life']),
+                                               namespace=config['namespace'],
+                                               sender=config['sender'])
+            configs.append(condition_config)
+        return configs
+
+    def handle_event(self, event: BaseEvent, task_state: object) -> Tuple[SchedulingAction, object]:
+        configs: List[ConditionConfig] = AIFlowHandler.parse_configs(self.config)
+        if task_state is None:
+            task_state = AiFlowTs()
+        af_ts = copy.deepcopy(task_state)
+        af_ts.event_map[(event.namespace, event.event_type, event.sender, event.key)] = event
```

Comment:
I don't understand here, what will happen if user parse a task_state which is not an AiFlowTs() ?

Reply:
AiFlowTs is created in the handler and stored in db. So the reading is consistent。